docs/csharp/language-reference/keywords/using-statement.md

```diff
@@ -39,7 +39,7 @@ Provides a convenient syntax that ensures the correct use of <xref:System.IDispo
 ## Remarks  
  <xref:System.IO.File> and <xref:System.Drawing.Font> are examples of managed types that access unmanaged resources (in this case file handles and device contexts). There are many other kinds of unmanaged resources and class library types that encapsulate them. All such types must implement the <xref:System.IDisposable> interface.  
   
- As a rule, when you use an `IDisposable` object, you should declare and instantiate it in a `using` statement. The `using` statement calls the <xref:System.IDisposable.Dispose%2A> method on the object in the correct way, and (when you use it as shown earlier) it also causes the object itself to go out of scope as soon as <xref:System.IDisposable.Dispose%2A> is called. Within the `using` block, the object is read-only and cannot be modified or reassigned.  
+When the lifetime of an `IDisposable` object is limited to a single method, you should declare and instantiate it in a `using` statement. The `using` statement calls the <xref:System.IDisposable.Dispose%2A> method on the object in the correct way, and (when you use it as shown earlier) it also causes the object itself to go out of scope as soon as <xref:System.IDisposable.Dispose%2A> is called. Within the `using` block, the object is read-only and cannot be modified or reassigned.  
```

Comment:
I would change the word "**should**" to "**can**" in the sentence below:

> When the lifetime of an `IDisposable` object is limited to a single method, you **should** declare and instantiate it in a `using` statement.

While true for the most basic usage, such as the examples given in this article, things start to get a little more complicated when you start using multiple instances of IDisposables that interact with each other.

`using (StreamReader sr = new StreamReader(new FileStream("C:\myfile.txt", FileMode.Create)))  
{ ... }`

will generate a [CA2000](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca2000-dispose-objects-before-losing-scope) warning if Code Analysis is turned on.

Another example taken from the sample code on the [CryptoStream Class](https://docs.microsoft.com/en-us/dotnet/api/system.security.cryptography.cryptostream?view=netframework-4.7) documentation... 

    using (Rijndael rijAlg = Rijndael.Create())
    {
        rijAlg.Key = Key;
        rijAlg.IV = IV;

        // Create a decryptor to perform the stream transform.
        ICryptoTransform decryptor = rijAlg.CreateDecryptor(rijAlg.Key, rijAlg.IV);

        // Create the streams used for decryption.
        using (MemoryStream msDecrypt = new MemoryStream(cipherText))
        {
            using (CryptoStream csDecrypt = new CryptoStream(msDecrypt, decryptor, CryptoStreamMode.Read))
            {
                using (StreamReader srDecrypt = new StreamReader(csDecrypt))
                {

                    // Read the decrypted bytes from the decrypting stream
                    // and place them in a string.
                    plaintext = srDecrypt.ReadToEnd();
                }
            }
        }

    }



This code as is will generate [CA2202](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca2202-do-not-dispose-objects-multiple-times) warnings when Code Analysis is turned on.

**Not directly related to this commit:**
- A future improvement for the article would be to include examples (as those shown here) on when not to use the `using` statement and what should be done instead (try/finally for instance)

Reply:
I recommend keeping "should" over "can". In the general case, you **should** put declarations in a `using` statement.

There are exceptions, and I recommend adding a disclaimer that there are exceptions, and link to the two references @alfredmyers points out.

Thoughts?

Reply:
I have to agree that "can" isn't the best choice here, but neither is "should".

There are several cases where Dispose doesn't have to be called at all. The fact is, `IDisposable` is a leaky abstraction. Pun intended.

Take for instance `Task`. It does implement IDisposable, but calling Dispose is only recommended in certain circumstances. See [Do I need to dispose of Tasks?](https://blogs.msdn.microsoft.com/pfxteam/2012/03/25/do-i-need-to-dispose-of-tasks/)

The same could be said about `DataSet` for which IDisposable was implemented for the sake of supporting the visual designer back in the good ol'days, but wasn't needed at all in other situations.

I like the the terms `Do`, `Consider`, `Avoid`, and `Do not` used by the [Framework Design Guidelines](https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/). Borrowing from it, I would tweak the text a little bit more to

> When the lifetime of an `IDisposable` object is limited to a single method, consider declaring and instantiating it in a `using` statement.